scripts/gc.js

```diff
@@ -0,0 +1,433 @@
+"use strict";
+Object.defineProperty(exports, "__esModule", { value: true });
+const rainblock = require("@rainblock/merkle-patricia-tree");
+const ethereumjs = require("merkle-patricia-tree");
+const stream_chain_1 = require("stream-chain");
+const stream_json_1 = require("stream-json");
+const Pick_1 = require("stream-json/filters/Pick");
+const StreamObject_1 = require("stream-json/streamers/StreamObject");
+const path = require("path");
+const fs = require("fs-extra");
+const asyncChunks = require('async-chunks');
+const RLP = require('rlp');
+const ethUtil = require('ethereumjs-util');
+const benchmark = require('benchmark');
+const async = require('async');
+const {promisify} = require('util');
+const memwatch = require('node-memwatch');
+const wait = require('wait-for-stuff');
+
+const heapdump = require('heapdump');
+const BranchNode = require("@rainblock/merkle-patricia-tree/build/src/index").BranchNode;
+const LeafNode = require("@rainblock/merkle-patricia-tree/build/src/index").LeafNode;
+const ExtensionNode = require("@rainblock/merkle-patricia-tree/build/src/index").ExtensionNode;
+
+function ethereumAccountToRlp(account) {
+    let hexBalance = BigInt(`${account.balance}`).toString(16);
+    if (hexBalance === '0') {
+        hexBalance = '';
+    }
+    else if (hexBalance.length % 2 === 1) {
+        hexBalance = `0${hexBalance}`;
+    }
+    return RLP.encode([
+        account.nonce,
+        Buffer.from(hexBalance, 'hex'),
+        Buffer.from(account.root, 'hex'),
+        Buffer.from(account.codeHash, 'hex')
+    ]);
+}
+
+function delay(t, val) {
+   return new Promise(function(resolve) {
+       setTimeout(function() {
+         resolve(val);
+       }, t);
+   });
+}
+
+const readFile = async (maxKeys) => {
+  const blockNum = process.argv[2];
+  const filename = path.join(__dirname + '../../../stateDumps/state_' + blockNum +'.json');
+  const pipeline = stream_chain_1.chain([
+    fs.createReadStream(filename),
+    stream_json_1.parser(),
+    Pick_1.pick({ filter: 'accounts' }),
+    StreamObject_1.streamObject(),
+  ]);
+
+   const els = [];
+   const elt = [];
+   let numKeys = 0;
+   for await (const data of asyncChunks(pipeline)) {
+     if (numKeys < maxKeys) {
+       const key = ethUtil.keccak256(Buffer.from(data.key, 'hex'));
+       const val = (ethereumAccountToRlp(data.value));
+       els.push({'key': key, 'val': val});
+       numKeys += 1;
+     }
+   }
+  return els;
+}
```

Comment:
Line: 49 - 71
Please use the utils.readStateDump function which returns a pipeline.
Read the required number of keys from the pipeline and ignore the rest.


- Helps to avoid code duplication.